source/node_test.go

```diff
@@ -93,123 +97,146 @@ func testNodeSourceEndpoints(t *testing.T) {
+			title:              "node with both external and internal IP returns an endpoint with external IP",
+			nodeName:           "node1",
+			exposeInternalIPv6: true,
+			nodeAddresses:      []v1.NodeAddress{{Type: v1.NodeExternalIP, Address: "1.2.3.4"}, {Type: v1.NodeInternalIP, Address: "2.3.4.5"}},
 			expected: []*endpoint.Endpoint{
 				{RecordType: "A", DNSName: "node1", Targets: endpoint.Targets{"1.2.3.4"}},
 			},
 		},
 		{
-			title:         "node with both external, internal, and IPv6 IP returns endpoints with external IPs",
-			nodeName:      "node1",
-			nodeAddresses: []v1.NodeAddress{{Type: v1.NodeExternalIP, Address: "1.2.3.4"}, {Type: v1.NodeInternalIP, Address: "2.3.4.5"}, {Type: v1.NodeInternalIP, Address: "2001:DB8::8"}},
+			title:              "node with both external, internal, and IPv6 IP returns endpoints with external IPs",
+			nodeName:           "node1",
+			exposeInternalIPv6: true,
+			nodeAddresses:      []v1.NodeAddress{{Type: v1.NodeExternalIP, Address: "1.2.3.4"}, {Type: v1.NodeInternalIP, Address: "2.3.4.5"}, {Type: v1.NodeInternalIP, Address: "2001:DB8::8"}},
 			expected: []*endpoint.Endpoint{
 				{RecordType: "A", DNSName: "node1", Targets: endpoint.Targets{"1.2.3.4"}},
 				{RecordType: "AAAA", DNSName: "node1", Targets: endpoint.Targets{"2001:DB8::8"}},
 			},
 		},
 		{
-			title:         "node with only internal IP returns an endpoint with internal IP",
-			nodeName:      "node1",
-			nodeAddresses: []v1.NodeAddress{{Type: v1.NodeInternalIP, Address: "2.3.4.5"}},
+			title:              "node with only internal IP returns an endpoint with internal IP",
+			nodeName:           "node1",
+			exposeInternalIPv6: true,
+			nodeAddresses:      []v1.NodeAddress{{Type: v1.NodeInternalIP, Address: "2.3.4.5"}},
 			expected: []*endpoint.Endpoint{
 				{RecordType: "A", DNSName: "node1", Targets: endpoint.Targets{"2.3.4.5"}},
 			},
 		},
 		{
-			title:         "node with only internal IPs returns endpoints with internal IPs",
-			nodeName:      "node1",
-			nodeAddresses: []v1.NodeAddress{{Type: v1.NodeInternalIP, Address: "2.3.4.5"}, {Type: v1.NodeInternalIP, Address: "2001:DB8::8"}},
+			title:              "node with only internal IPs returns endpoints with internal IPs",
+			nodeName:           "node1",
+			exposeInternalIPv6: true,
+			nodeAddresses:      []v1.NodeAddress{{Type: v1.NodeInternalIP, Address: "2.3.4.5"}, {Type: v1.NodeInternalIP, Address: "2001:DB8::8"}},
 			expected: []*endpoint.Endpoint{
 				{RecordType: "A", DNSName: "node1", Targets: endpoint.Targets{"2.3.4.5"}},
 				{RecordType: "AAAA", DNSName: "node1", Targets: endpoint.Targets{"2001:DB8::8"}},
 			},
 		},
 		{
-			title:         "node with neither external nor internal IP returns no endpoints",
-			nodeName:      "node1",
-			nodeAddresses: []v1.NodeAddress{},
-			expectError:   true,
+			title:              "node with only internal IPs with expose internal IP as false shouldn't return AAAA endpoints with internal IPs",
```

Comment:
In case there are ONLY internal IPs, all `InternalIP`s should be exposed. No matter if they are IPv6 or IPv4.
This was the behavior before #3588 was merged. See [ref](https://github.com/kubernetes-sigs/external-dns/pull/3588/files#diff-6b0927f853e71f06b56639a526f0bff75b0df856034022444fa573af909b91e4L6)

Reply:
@jonasbadstuebner / @ivankatliarchuk so even if the flag is set as true we expose those IPs? That wasn't in the proposal.. I can change the behavior and add a test case for that.. let me know about the same.

Reply:
Yes, otherwise it would break setup of some users, where the providers only set this IP.